### PR TITLE
feat(documentation) Adds a small paragraph describing how to use another timezone

### PR DESCRIPTION
**Pull request by gounthar**
By default, the image is using `Etc/UTC`. A user wants to use another timezone (c.f. [#291](https://github.com/jenkinsci/docker-inbound-agent/issues/291).

I explain briefly how to use another one while creating a new image derived from `jenkins/agent`.

This is a follow-up to #337 where I added the `tzdata` package.
This PR and the #337 will replace my first attempt (see #334).

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- ~~Ensure you have provided tests - that demonstrates feature works or fixes the issue~~

### CODE REVIEW

**Comment by dduportal** [comment]
Not blocking, but maybe suggesting mounting the host `/etc/localtime` along with setting the `TZ` environment variable (as per https://github.com/moby/moby/issues/12084#issuecomment-89697533) could also be useful, to cover users who do not want to build a custom image.

For instance, from the #337  (with tzdata installed)

```
$ docker build -t test ./11/bullseye && docker run --rm -ti --entrypoint=date -v /etc/localtime:/etc/localtime:ro test && docker run --rm -ti --env TZ=Asia/Shanghai --entrypoint=date test 
[+] Building 2.6s (16/16) FINISHED                                                                                                                                                                                                        
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 32B                                                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-20221114                                                                                                                                                          1.8s
 => [internal] load metadata for docker.io/library/eclipse-temurin:11.0.17_8-jdk-focal                                                                                                                                               1.9s
 => [stage-1 1/8] FROM docker.io/library/debian:bullseye-20221114@sha256:3066ef83131c678999ce82e8473e8d017345a30f5573ad3e44f62e5c9c46442b                                                                                            0.0s
 => [jre-build 1/2] FROM docker.io/library/eclipse-temurin:11.0.17_8-jdk-focal@sha256:d84b470fe0b8de379461a8e179792c389e29611b6a4f182c02c23b98c2527f7e                                                                               0.0s
 => https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/3077.vd69cf116da_6f/remoting-3077.vd69cf116da_6f.jar                                                                                                             0.5s
 => CACHED [stage-1 2/8] RUN groupadd -g "1000" "jenkins"   && useradd -l -c "Jenkins user" -d /home/jenkins -u "1000" -g "1000" -m "jenkins"                                                                                        0.0s
 => CACHED [stage-1 3/8] RUN apt-get update   && apt-get --yes --no-install-recommends install     ca-certificates     curl     fontconfig     git     git-lfs     less     netbase     openssh-client     patch     tzdata   && ap  0.0s
 => CACHED [stage-1 4/8] ADD --chown=jenkins:jenkins https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/3077.vd69cf116da_6f/remoting-3077.vd69cf116da_6f.jar /usr/share/jenkins/agent.jar                               0.0s
 => CACHED [stage-1 5/8] RUN chmod 0644 /usr/share/jenkins/agent.jar   && ln -sf /usr/share/jenkins/agent.jar /usr/share/jenkins/slave.jar                                                                                           0.0s
 => CACHED [jre-build 2/2] RUN jlink   --add-modules ALL-MODULE-PATH   --strip-debug   --no-man-pages   --no-header-files   --compress=2   --output /javaruntime                                                                     0.0s
 => CACHED [stage-1 6/8] COPY --from=jre-build /javaruntime /opt/java/openjdk                                                                                                                                                        0.0s
 => CACHED [stage-1 7/8] RUN mkdir /home/jenkins/.jenkins && mkdir -p "/home/jenkins/agent"                                                                                                                                          0.0s
 => CACHED [stage-1 8/8] WORKDIR /home/jenkins                                                                                                                                                                                       0.0s
 => exporting to image                                                                                                                                                                                                               0.0s
 => => exporting layers                                                                                                                                                                                                              0.0s
 => => writing image sha256:4ad90e28bb559cd34687da879b313bdb9c62c8af61b74834b655b962379d39e2                                                                                                                                         0.0s
 => => naming to docker.io/library/test                                                                                                                                                                                              0.0s
Fri Nov 25 15:52:10 UTC 2022
Fri Nov 25 23:52:11 CST 2022
```

and

```
 docker build -t test:alpine ./11/alpine && docker run --rm -ti --entrypoint=date -v /etc/localtime:/etc/localtime:ro test:alpine && docker run --rm -ti --env TZ=Asia/Shanghai --entrypoint=date test:alpine
[+] Building 1.0s (12/12) FINISHED                                                                                                                                                                                                        
 => [internal] load build definition from Dockerfile                                                                                                                                                                                 0.0s
 => => transferring dockerfile: 32B                                                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/eclipse-temurin:11.0.17_8-jdk-alpine                                                                                                                                              0.7s
 => [1/7] FROM docker.io/library/eclipse-temurin:11.0.17_8-jdk-alpine@sha256:dad17670f7d3e0072d2e3e04978dab261bd131bbd6117da57cc9d7c45b5029fa                                                                                        0.0s
 => https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/3077.vd69cf116da_6f/remoting-3077.vd69cf116da_6f.jar                                                                                                             0.1s
 => CACHED [2/7] RUN addgroup -g "1000" "jenkins"   && adduser -h /home/"jenkins" -u "1000" -G "jenkins" -D "jenkins"                                                                                                                0.0s
 => CACHED [3/7] RUN apk add --no-cache      bash      curl      git      git-lfs      musl-locales      openssh-client      openssl      procps      tzdata     && rm -rf /tmp/*.apk /tmp/gcc /tmp/gcc-libs.tar* /tmp/libz /tmp/li  0.0s
 => CACHED [4/7] ADD --chown=jenkins:jenkins https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/3077.vd69cf116da_6f/remoting-3077.vd69cf116da_6f.jar /usr/share/jenkins/agent.jar                                       0.0s
 => CACHED [5/7] RUN chmod 0644 /usr/share/jenkins/agent.jar   && ln -sf /usr/share/jenkins/agent.jar /usr/share/jenkins/slave.jar                                                                                                   0.0s
 => CACHED [6/7] RUN mkdir /home/"jenkins"/.jenkins && mkdir -p "/home/jenkins/agent"                                                                                                                                                0.0s
 => CACHED [7/7] WORKDIR /home/jenkins                                                                                                                                                                                               0.0s
 => exporting to image                                                                                                                                                                                                               0.0s
 => => exporting layers                                                                                                                                                                                                              0.0s
 => => writing image sha256:936f518194bb2fd614cf7c887d5f8bfe43de569d8acf50a7e1c91140b4596c51                                                                                                                                         0.0s
 => => naming to docker.io/library/test:alpine                                                                                                                                                                                       0.0s
Fri Nov 25 15:54:05 UTC 2022
Fri Nov 25 23:54:06 CST 2022
```

File: README.md [hunk 1]
```diff
@@ -79,3 +79,18 @@ The file `docker-bake.hcl` defines all the configuration for Linux images and th
...
+
+## Timezones
+
+By default, the image is using the `Etc/UTC` timezone.
+Should you want to adapt it to your local timezone while creating your own image based on `jenkins/agent`, you could use the following command (inspired by issue #[291](https://github.com/jenkinsci/docker-inbound-agent/issues/291)):
+
+```dockerfile
+FROM jenkins/agent as agent
+ [...]
+ENV TIME_ZONE=Asia/Shanghai
```
Comment (dduportal) [inline README.md:91]:
```suggestion
ENV TZ=Asia/Shanghai
```

File: README.md [hunk 2]
```diff
@@ -79,3 +79,18 @@ The file `docker-bake.hcl` defines all the configuration for Linux images and th
...
+ [...]
+RUN ln -snf /usr/share/zoneinfo/$TIME_ZONE /etc/localtime && echo $TIME_ZONE > /etc/timezone \
```
Comment (dduportal) [inline README.md:93]:
```suggestion
RUN ln -snf /usr/share/zoneinfo/"${TZ}" /etc/localtime && echo "${TZ}" > /etc/timezone \
```

File: README.md [hunk 3]
```diff
@@ -79,3 +79,39 @@ The file `docker-bake.hcl` defines all the configuration for Linux images and th
...
+## Timezones
+
+### Using directly the `jenkins/agent` image
+
+By default, the image is using the `Etc/UTC` timezone.
+If you want to use the timezone of your machine, you can mount the `/etc/localtime` file from the host (as per [this comment](https://github.com/moby/moby/issues/12084#issuecomment-89697533)) and the `/etc/timezone` from the host too.
+In this example, the machine is using the `Europe/Paris` timezone.
+
+```bash
+$ docker run --rm -ti --entrypoint=date -v /etc/localtime:/etc/localtime:ro -v /etc/timezone:/etc/timezone:ro jenkins/agent
```
Comment (dduportal) [inline README.md:92]:
```suggestion
$ docker run --rm --tty --interactive --entrypoint=date --volume=/etc/localtime:/etc/localtime:ro --volume=/etc/timezone:/etc/timezone:ro jenkins/agent
```

nitpick: use long flags to make the command easier to understand for the reader

File: README.md [hunk 4]
```diff
@@ -79,3 +79,39 @@ The file `docker-bake.hcl` defines all the configuration for Linux images and th
...
+Fri Nov 25 18:27:22 CET 2022
+```
+
+You can also set the `TZ` environment variable to the desired timezone.
+`TZ` is a standard POSIX environment variable used by many images, see [Wikipedia](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) for a list of valid values.
+The next command is run on a machine using the `Europe/Paris` timezone a few seconds after the previous one.
+
+```bash
+$ docker run --rm -ti --env TZ=Asia/Shanghai --entrypoint=date jenkins/agent
```
Comment (dduportal) [inline README.md:101]:
```suggestion
$ docker run --rm --tty --interactive --env TZ=Asia/Shanghai --entrypoint=date jenkins/agent
```

nitpick: use long flags to make the command easier to understand for the reader

**Review by dduportal — APPROVED** [review]
2 nitpicks suggestion feel free to accept or refuse them.

Otherwise it's nice, thanks!

File: README.md (hunk 3 above)
Comment (gounthar) [reply README.md:92]: That's a very good suggestion, sometimes, less is more, and sometimes, more is better. :-D

**Comment by gounthar** [comment]
Thanks a lot for the review, I always forget about those long flags. 😊

**Review by dduportal — APPROVED** [review]

**Pull request merged**